### ISSUE TITLE
System.NotImplementedException when setting the CRC FtpHashAlgorithm to client

### ISSUE THREAD

**Issue by PhantoManresa** [robinrodricks/FluentFTP#680]
**FTP OS:** I don't know

**FTP Server:** Titan FTP Server 17.00.3095

**Computer OS:** Windows 10

**FluentFTP Version:** 33.0.3

**Problem Description:**

When I try to set the Hash Algorithm of the server I get the following exception: 
**System.NotImplementedException**: 'The hash algorithm CRC was not advertised by the server.'

But if I access the server with WinSCP, it is capable of giving me a CRC32 checksum of the files so I assume that the FTP server supports it.

**Logs :**

```

# Connect()
Status:   Connecting to ***:21
Response: 220 Titan FTP Server 17.00.3095 Ready.
Command:  USER ***
Response: 331 User name okay, need password.
Command:  PASS ***
Response: 230-Welcome *** from ***. You are now logged in to the server.
Response: 230 User logged in, proceed.
Command:  FEAT
Response: 211-Extensions Supported
Response: COMB
Response: MLST type*;size*;modify*;create*;perm*;
Response: UTF8
Response: SIZE
Response: MDTM
Response: MFMT
Response: MFCT
Response: EPRT
Response: EPSV
Response: XCRC
Response: REST STREAM
Response: CCC
Response: PBSZ
Response: PROT
Response: MODE Z
Response: 211 End
Status:   Text encoding: System.Text.UTF8Encoding+UTF8EncodingSealed
Command:  OPTS UTF8 ON
Response: 200 UTF-8 Encoding enabled.
Command:  SYST
Response: 215 UNIX Type: L8

```

**Comment by robinrodricks** [comment]
Which hashing method are you using? Can you try with the XCRC method and report back?

**Comment by PhantoManresa** [comment]
I was trying to use `SetHashAlgorithm(FtpHashAlgorithm.CRC)` 

If I use the GetXCRC method, it throws `FluentFTP.FtpCommandException: Command not understood.`

**Comment by robinrodricks** [comment]
FtpCommandException will simply throw with the text received by the server. Perhaps the server does not support `XCRC` - try with GetCRC or something else? That's wierd because your server reports that it understands `XCRC` in the `FEAT` response, so its possible its a server bug. Worst case you can try `client.Execute("XCRC [filepath]")` and see if you get a response or not, that way we know if its a library bug or server bug.

**Comment by PhantoManresa** [comment]
Ok, I just realized that when I tried to use the `GetXCRC` method I did it in the wrong server. A foolish mistake. I did it in the right server this time and it returns the hash properly, so the command is not the problem. But if I try to set the hash algorithm via `SetHashAlgorithm(FtpHashAlgorithm.CRC)`, then I have the problem described in the original post.

**Comment by robinrodricks** [comment]
Fixed by simplifying the API in 35.0.0.

**Comment by robinrodricks** [comment]
Marking this as incorrect usage since you don't need to `SetHashAlgorithm` if you are using XCRC commands. Its only needed for HASH command.
